aws_google_auth/configuration.py

```diff
@@ -0,0 +1,144 @@
+
+        # profile
+        assert (self.profile.__class__ is str), "Expected profile to be a string. Got {}.".format(self.profile.__class__)
+
+        # region
+        assert (self.region.__class__ is str), "Expected region to be a string. Got {}.".format(self.region.__class__)
+
+        # idp_id
+        assert (self.idp_id is not None), "Expected idp_id to be set to non-None value."
+
+        # sp_id
+        assert (self.sp_id is not None), "Expected sp_id to be set to non-None value."
+
+        # username
+        assert (self.username.__class__ is str), "Expected username to be a string. Got {}.".format(self.username.__class__)
+
+        # password
+        assert (self.password.__class__ is str), "Expected password to be a string. Got {}.".format(self.password.__class__)
+
+        # role_arn (Can be blank, we'll just prompt)
+        if self.role_arn is not None:
+            assert (self.role_arn.__class__ is str), "Expected role_arn to be None or a string. Got {}.".format(self.role_arn.__class__)
+            assert ("arn:aws:iam::" in self.role_arn), "Expected role_arn to contain 'arn:aws:iam::'. Got '{}'.".format(self.role_arn)
+
+        # u2f_disabled
+        assert (self.u2f_disabled.__class__ is bool), "Expected u2f_disabled to be a boolean. Got {}.".format(self.u2f_disabled.__class__)
+
+    # Write the configuration (and credentials) out to disk. This allows for
+    # regular AWS tooling (aws cli and boto) to use the credentials in the
+    # profile the user specified.
+    def write(self, amazon_object):
+        self.ensure_config_files_exist()
+
+        assert (self.profile is not None), "Can not store config/credentials if the AWS_PROFILE is None."
+
+        # Write to the configuration file
+        config_parser = configparser.RawConfigParser()
+        config_parser.read(self.config_file)
+        if not config_parser.has_section(self.profile):
+            config_parser.add_section(self.profile)
+        config_parser.set(self.profile, 'region', self.region)
+        config_parser.set(self.profile, 'google_config.ask_role', self.ask_role)
+        config_parser.set(self.profile, 'google_config.duration', self.duration)
+        config_parser.set(self.profile, 'google_config.idp_id', self.idp_id)
+        config_parser.set(self.profile, 'google_config.role_arn', self.role_arn)
+        config_parser.set(self.profile, 'google_config.sp_id', self.sp_id)
+        config_parser.set(self.profile, 'google_config.u2f_disabled', self.u2f_disabled)
+        config_parser.set(self.profile, 'google_config.username', self.username)
+        with open(self.config_file, 'w+') as f:
+            config_parser.write(f)
+
+        # Write to the credentials file (only if we have credentials)
+        if amazon_object is not None:
+            credentials_parser = configparser.RawConfigParser()
+            credentials_parser.read(self.credentials_file)
+            if not credentials_parser.has_section(self.profile):
```

Comment:
Will need to add the `profile` prefix to the non `default` profiles too.

Reply:
Fixed in 00c97787b55c0401a450a2f3a1207f720c5d093b.